src/DropdownButton.jsx

```diff
@@ -111,6 +111,11 @@ var DropdownButton = React.createClass({
     e.preventDefault();
 
     this.setDropdownState(!this.state.open);
+    if(this.props.onClick && this.props.noCaret ){
```

Comment:
I found a scenario where my DropDownButton didn't have any items (maybe
related to user permissions) then I had to use noCaret=true but I needed to
keep working with onClick handler

2015-04-13 16:30 GMT+02:00 Mathias Schreck <notifications@github.com>:

> In src/DropdownButton.jsx
> <https://github.com/react-bootstrap/react-bootstrap/pull/458#discussion_r28242327>
> :
>
> > @@ -111,6 +111,11 @@ var DropdownButton = React.createClass({
> >      e.preventDefault();
> >
> >      this.setDropdownState(!this.state.open);
> > +    if(this.props.onClick && this.props.noCaret ){
>
> Why should noCaret be set to true in order to execute the onClick handler?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/react-bootstrap/react-bootstrap/pull/458/files#r28242327>
> .
>